patricia/patricia_dense_test.go

```diff
@@ -197,15 +199,18 @@ func TestTrie_DeleteLeakageDense(t *testing.T) {
 
 	oldBytes := heapAllocatedBytes()
 
-	for _, v := range data {
-		if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
-			t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
+	// repeat insertion/deletion for 10K times to catch possible memory issues
+	for i := 0; i < 10000; i++ {
+		for _, v := range data {
+			if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
```

Comment:
it inserts it, then immediately deletes it before going into the next loop iteration - both "insert" and "delete" for loops are inside of the 10K repetition one.  If there is a memory leak, shouldn't that leak be amplified 10000 times and thus get to the noticeable magnitude beyond the GC "noise"?  Sorry if I'm missing a point here...

```
	for i := 0; i < 10000; i++ {
		for _, v := range data {
			if ok := trie.Insert(Prefix(v.key), v.value); ok != v.retVal {
				t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
			}
		}

		for _, v := range data {
			if ok := trie.Delete([]byte(v.key)); ok != v.retVal {
				t.Errorf("Unexpected return value, expected=%v, got=%v", v.retVal, ok)
			}
		}
```